File: 3522-find-the-power-of-k-size-subarrays-i/find-the-power-of-k-size-subarrays-i.java
class Solution {

    public int[] resultsArray(int[] nums, int k) 
    {
        int length = nums.length;
        int[] result = new int[length - k + 1];
        Deque<Integer> indexDeque = new ArrayDeque<>();
        for (int currentIndex = 0; currentIndex < length; currentIndex++) 
        {
            // Check if current element breaks the consecutive and sorted condition
            if(!indexDeque.isEmpty() &&nums[currentIndex] != nums[currentIndex - 1] + 1)
            indexDeque.clear(); // Invalidate the entire deque if condition breaks

            // Add current element index to the deque
            indexDeque.offerLast(currentIndex);

            // Check if the window is of size k and update result
            if (currentIndex >= k - 1) 
            {
                // Valid window of size >= k
                if(indexDeque.size() >= k)  
                result[currentIndex - k + 1] = nums[indexDeque.peekLast()];

                // Not valid, return -1
                else
                result[currentIndex - k + 1] = -1; 

                // Remove elements that are out of the window
                if(!indexDeque.isEmpty() && indexDeque.peekFirst() < currentIndex - k + 1) 
                indexDeque.pollFirst();
            }
        }

        return result;
    }
}

File: 3522-find-the-power-of-k-size-subarrays-i/find-the-power-of-k-size-subarrays-i.cpp
class Solution {
public:
    vector<int> resultsArray(vector<int>& nums, int k) 
    {
        if(k == 1)
        return nums;
        
        int n = nums.size();
        vector<int> res(n - k + 1);
        int count = 1, i = 0;
        for(int j = 1; j < n; j++)
        {
            if(nums[j] == nums[j - 1] + 1)
            count++;
            else
            count = 1;

            if(j - i + 1 == k)
            {
                if(count >= k)
                res[i++] = nums[j];

                else
                res[i++] = -1;
            }
        }
        return res;
    }
};